mailpoet/lib/Automation/Engine/Builder/UpdateWorkflowController.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Builder;
+
+use MailPoet\Automation\Engine\Exceptions;
+use MailPoet\Automation\Engine\Storage\WorkflowStorage;
+use MailPoet\Automation\Engine\Workflows\Workflow;
+
+class UpdateWorkflowController {
+  /** @var WorkflowStorage */
+  private $storage;
+
+  public function __construct(
+    WorkflowStorage $storage
+  ) {
+    $this->storage = $storage;
+  }
+
+  public function updateWorkflow(int $id, array $data): Workflow {
+    // TODO: data & workflow validation (trigger existence, graph consistency, etc.)
+    // TODO: new revisions when content is changed
+    // TODO: validation when status being is changed
+
+    $workflow = $this->storage->getWorkflow($id);
+    if (!$workflow) {
+      throw Exceptions::workflowNotFound($id);
+    }
+
+    if (array_key_exists('status', $data)) {
+      $workflow->setStatus($data['status']);
```

Comment:
I assume `status` is an enum and would get validated in the `ToDo` section above?

Reply:
It gets somewhat validated [here](https://github.com/mailpoet/mailpoet/pull/4220/files/48599edd5c8a0e021d20ce1d9cc12a7ee754e85b#diff-7d33db3284a65610c0bcd8679866b78534f97a059905da0df3a2373db24dafe0R35) but we don't have enums yet. Do you think we should validate it also inside the controller to be on the safe side? 🤔

I was also considering whether we should make the status change a dedicated endpoint (like `/workflow/<id>/status`) but looking at WP posts they don't do such a thing so I kept it this way for now.

Reply:
> Whether we should make the status change a dedicated endpoint

I think this is not needed. The general entity endpoint would want to be able to `PUT status:active` anyway imo (in case you later change the UI to a activate toggle and a save button, which would save all changes). To have a second endpoint for just the status seems not necessary for me. Besides what would be the payload? only `"active"|"inactive"`?

Since we do not have `enums` (though, I would like that) I think a simple
```
$workflow->setStatus($data['status'] === 'active' ? 'active':'inactive');
```
would be nice.

Reply:
@websupporter [Enums are available from WP 5.7](https://github.com/WordPress/WordPress/blob/e70ac7004c8773249ec47fc56a73953043a6a877/wp-includes/rest-api.php#L1981) so I guess we may as well add them soon and use them in for automation and once we release the MVP, we'd have to bump minimum WP version.

As for this PR, I can just add a check that an existing value is being set for now.

Reply:
That sounds good. Thank you!

Reply:
I added this: https://github.com/mailpoet/mailpoet/commit/cf752545d339de14804252603661de9f3f53d6b7

Reply:
Thank you!